isisd/isis_main.c

```diff
@@ -196,6 +196,93 @@ FRR_DAEMON_INFO(isisd, ISIS, .vty_port = ISISD_VTY_PORT,
 		.n_yang_modules = array_size(isisd_yang_modules),
 );
 
+#ifdef FUZZING
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+static bool FuzzingInit(void)
+{
+	const char *name[] = {"isisd"};
+
+#ifdef FABRICD
+	frr_preinit(&fabricd_di, 1, (char **)name);
+#else
+	frr_preinit(&isisd_di, 1, (char **)name);
+#endif
+
+	/* INIT */
+	isis_master_init(frr_init_fast());
+
+	isis_error_init();
+	access_list_init();
+	access_list_add_hook(isis_filter_update);
+	access_list_delete_hook(isis_filter_update);
+	isis_vrf_init();
+	prefix_list_init();
+	prefix_list_add_hook(isis_prefix_list_update);
+	prefix_list_delete_hook(isis_prefix_list_update);
+	isis_init();
+	isis_circuit_init();
+	isis_spf_init();
+	isis_route_map_init();
+	isis_mpls_te_init();
+	isis_sr_init();
+	lsp_init();
+	isis_zebra_init(master, 1);
+	isis_bfd_init(master);
+
+	return true;
+}
+
+static struct isis_circuit *FuzzingCreateIsis(void)
+{
+	struct prefix p;
+	struct interface *ifp = NULL;
+	struct connected *conn = NULL;
+	struct isis_area *area = NULL;
+	struct isis_circuit *c = NULL;
+
+	ifp = if_create_ifindex(69, 0);
+	ifp->mtu = 68;
+	str2prefix("11.0.2.2/24", &p);
+
+	conn = connected_add_by_prefix(ifp, &p, NULL);
+
+	c = isis_circuit_new(ifp, "fuzz-tag");
+
+	area = isis_area_create("FUZZ_AREA", NULL);
+	isis_area_add_circuit(area, c);
+
+	return c;
+}
+
+static struct isis_circuit *FuzzingIsis;
+static bool FuzzingInitialized;
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+	struct isis_circuit *c;
+	static uint8_t ssnpa[6] = {0x2c, 0x54, 0x91, 0x88, 0xc9, 0xe3};
```

Comment:
Perhaps a brief comment on what this does / is could help others?

Reply:
Added a comment here and changed the MAC address to one that falls into the IANA documentation range as I'm not sure where I got the original address from.
There's also another comment added to explain why `isis_handle_pdu` is used.